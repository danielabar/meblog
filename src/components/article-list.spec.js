/**
 * @jest-environment jsdom
 */

import React from "react"
import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"

import ArticleList from "./article-list"

describe("AllLink", () => {
  it("renders correctly", () => {
    const testData = [
      {
        node: {
          id: "123abc",
          fields: {
            slug: "/blog/some-article-1",
          },
          frontmatter: {
            title: "Some Article 1",
            category: "Rails",
            date: "August 2021",
          },
          excerpt: "First sentence of article 1 blah blah blah...",
        },
      },
      {
        node: {
          id: "456def",
          fields: {
            slug: "/blog/some-article-2",
          },
          frontmatter: {
            title: "Some Article 2",
            category: "Docker",
            date: "September 2021",
          },
          excerpt: "First sentence of article 2 blah blah blah...",
        },
      },
    ]

    const container = render(<ArticleList articles={testData} />)
    expect(container).toMatchSnapshot()

    expect(screen.getByText("Some Article 1")).toBeInTheDocument()
    expect(screen.getByText("Some Article 2")).toBeInTheDocument()
  })
})
