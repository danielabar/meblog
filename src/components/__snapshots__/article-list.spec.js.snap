// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="container"
        data-testid="article-list"
      >
        <article
          class="container"
        >
          <div
            class="subheader"
          >
            <div
              class="date"
            >
              August 2021
            </div>
            <div
              class="category"
            >
              Rails
            </div>
          </div>
          <a
            class="title"
            href="/blog/some-article-1"
          >
            Some Article 1
          </a>
          <div
            class="excerpt"
          >
            First sentence of article 1 blah blah blah...
          </div>
          <a
            class="read"
            href="/blog/some-article-1"
          >
            Read
          </a>
        </article>
        <article
          class="container"
        >
          <div
            class="subheader"
          >
            <div
              class="date"
            >
              September 2021
            </div>
            <div
              class="category"
            >
              Docker
            </div>
          </div>
          <a
            class="title"
            href="/blog/some-article-2"
          >
            Some Article 2
          </a>
          <div
            class="excerpt"
          >
            First sentence of article 2 blah blah blah...
          </div>
          <a
            class="read"
            href="/blog/some-article-2"
          >
            Read
          </a>
        </article>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="container"
      data-testid="article-list"
    >
      <article
        class="container"
      >
        <div
          class="subheader"
        >
          <div
            class="date"
          >
            August 2021
          </div>
          <div
            class="category"
          >
            Rails
          </div>
        </div>
        <a
          class="title"
          href="/blog/some-article-1"
        >
          Some Article 1
        </a>
        <div
          class="excerpt"
        >
          First sentence of article 1 blah blah blah...
        </div>
        <a
          class="read"
          href="/blog/some-article-1"
        >
          Read
        </a>
      </article>
      <article
        class="container"
      >
        <div
          class="subheader"
        >
          <div
            class="date"
          >
            September 2021
          </div>
          <div
            class="category"
          >
            Docker
          </div>
        </div>
        <a
          class="title"
          href="/blog/some-article-2"
        >
          Some Article 2
        </a>
        <div
          class="excerpt"
        >
          First sentence of article 2 blah blah blah...
        </div>
        <a
          class="read"
          href="/blog/some-article-2"
        >
          Read
        </a>
      </article>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
